### PR TITLE
Abstract the drag dismiss to use the new library

### PR DESCRIPTION
**Pull request by klinker24**
- Update to v0.20.0

This also came with a few other (almost) freebies:
- Use `DragDismissActivity` on the image viewer
- Add pitch black theme option

---

After merging, we should push to maven and update the build number in the readme

### CODE REVIEW

File: library/src/main/java/xyz/klinker/android/article/ArticleIntent.java [hunk 1]
```diff
@@ -275,6 +278,11 @@
     public static final int THEME_AUTO = 3;
 
     /**
+     * Use an automatic theme, changing between light and dark based on time of day.
+     */
+    public static final int THEME_BLACK = 4;
```
Comment (klinker41) [inline library/src/main/java/xyz/klinker/android/article/ArticleIntent.java]: Can we switch this and theme auto for better organization?

File: library/src/main/java/xyz/klinker/android/article/ArticleIntent.java [hunk 2]
```diff
@@ -578,6 +589,15 @@ public ArticleIntent build() {
                 mIntent.putParcelableArrayListExtra(EXTRA_TOOLBAR_ITEMS, mActionButtons);
             }
             mIntent.putExtra(EXTRA_ENABLE_INSTANT_APPS, mInstantAppsEnabled);
+
+            new DragDismissIntentBuilder(context)
+                    .setShouldScrollToolbar(true)
+                    .setShowToolbar(true)
```
Comment (klinker41) [inline library/src/main/java/xyz/klinker/android/article/ArticleIntent.java]: Can you switch this and line 595?

File: library/src/main/java/xyz/klinker/android/article/ArticleIntent.java [hunk 3]
```diff
@@ -616,4 +636,24 @@ public static boolean shouldAlwaysUseBrowserUI(Intent intent) {
...
+    /**
+     * Converts the ArticleIntent integer theme to a DragDismiss theme
+     *
+     * @param themeInt One of {@link #THEME_LIGHT}, {@link #THEME_DARK}, {@link #THEME_BLACK} or
+     *                  {@link #THEME_AUTO}.
+     * @return the corresponding DragDismiss theme
+     */
+    static DragDismissIntentBuilder.Theme convertIntToTheme(int themeInt) {
+        DragDismissIntentBuilder.Theme theme = DragDismissIntentBuilder.Theme.DAY_NIGHT;
+        if (themeInt == THEME_LIGHT) {
```
Comment (klinker41) [inline library/src/main/java/xyz/klinker/android/article/ArticleIntent.java]: Make this a switch statement and return in line please.

**Comment by klinker24** [comment]
That is? Must have done a pretty good job then ;)

Done with those changes. I think I removed everything I needed too (classes, methods). Make sure, then merge when ready

**Comment by klinker41** [comment]
That was just my first pass. Can you increment the build number too? I'll test it tomorrow night and then merge.

File: library/src/main/java/xyz/klinker/android/article/ArticleIntent.java [hunk 4]
```diff
@@ -272,7 +275,12 @@
     /**
      * Use an automatic theme, changing between light and dark based on time of day.
```
Comment (klinker41) [inline library/src/main/java/xyz/klinker/android/article/ArticleIntent.java]: Update this javadoc.

File: library/src/main/java/xyz/klinker/android/article/ArticleIntent.java [hunk 5]
```diff
@@ -616,4 +636,21 @@ public static boolean shouldAlwaysUseBrowserUI(Intent intent) {
...
+    /**
+     * Converts the ArticleIntent integer theme to a DragDismiss theme
+     *
+     * @param themeInt One of {@link #THEME_LIGHT}, {@link #THEME_DARK}, {@link #THEME_BLACK} or
+     *                  {@link #THEME_AUTO}.
+     * @return the corresponding DragDismiss theme
+     */
+    static DragDismissIntentBuilder.Theme convertIntToTheme(int themeInt) {
+        switch (themeInt) {
+            case THEME_LIGHT: return DragDismissIntentBuilder.Theme.LIGHT;
```
Comment (klinker41) [inline library/src/main/java/xyz/klinker/android/article/ArticleIntent.java]: Fix the formatting on these. Either bring the default up to the same line or move these down to the next line.

File: library/src/main/java/xyz/klinker/android/article/ImageViewActivity.java [hunk 6]
```diff
@@ -16,31 +16,41 @@
...
 import com.bumptech.glide.load.engine.DiskCacheStrategy;
 
+import xyz.klinker.android.drag_dismiss.activity.DragDismissActivity;
+
 /**
  * Activity for viewing full size images from an article. Images are zoomable. You can pass in the
  * image url from an intent with .setData(url).
  */
-public final class ImageViewActivity extends AppCompatActivity {
+public final class ImageViewActivity extends DragDismissActivity {
```
Comment (klinker41) [inline library/src/main/java/xyz/klinker/android/article/ImageViewActivity.java]: By doing this, we run into the problem that the image activity is not full screen on tablets, which is far from ideal. If we can't make it full screen then the image activity needs to go back to the old `AppCompatActivity`.

File: library/src/main/java/xyz/klinker/android/article/ArticleIntent.java (hunk 5 above)
Comment (klinker24) [inline library/src/main/java/xyz/klinker/android/article/ArticleIntent.java]: Fine fine

File: library/src/main/java/xyz/klinker/android/article/ImageViewActivity.java (hunk 6 above)
Comment (klinker24) [inline library/src/main/java/xyz/klinker/android/article/ImageViewActivity.java]: Yes, I did the same in talon and feel the same way. Should be fairly trivial to add this, will do

Comment (klinker24) [inline library/src/main/java/xyz/klinker/android/article/ImageViewActivity.java]: When 1.2.0 goes live on Maven, this should be done. It has just been pushed

**Pull request merged**